src/main/java/core/basesyntax/file/DataConverterImpl.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax.file;
+
+import core.basesyntax.service.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputReport) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        for (String line : inputReport) {
+            if (line.startsWith("type")) {
+                continue;
+            }
+            String[] parts = line.split(",");
+            String operationCode = parts[0];
+            String fruit = parts[1];
+            int quantity = Integer.parseInt(parts[2]);
```

Comment:
Consider adding error handling for `Integer.parseInt(parts[2])` to manage cases where the quantity is not a valid integer. This will prevent the application from crashing due to malformed input data.